test/functional/test_drive_capacity_consistency.py

```diff
@@ -0,0 +1,150 @@
+            "drives": [
+                {
+                    "size": 10,
+                    "model": "SATADOM",
+                    "serial": "20160518AA851134100",
+                    "file": fixtures.image
+                },
+                {
+                    "format": "raw",
+                    "size": 1,
+                    "vendor": "SEAGATE",
+                    "product": "ST4000NM0005",
+                    "serial": "01234567",
+                    "version": "M001",
+                    "wwn": "0x5000C500852E2971",
+                    "share-rw": "true",
+                    "cache": "none",
+                    "scsi-id": 1,
+                    "slot_number": 0,
+                    "sector_size": 520,
+                    "file": "/home/infrasim/.infrasim/test_drive_capacity.img"
+                 }
+            ]
+        }]
+    conf["compute"]["networks"] = [
+        {
+            "bus": "pcie.0",
+            "device": "e1000",
+            "mac": "52:54:be:b9:77:dd",
+            "network_mode": "nat",
+            "network_name": "dummy0",
+        }]
+    node = model.CNode(conf)
+    node.init()
+    node.precheck()
+    node.start()
+    node.wait_node_up(timeout=20)
+    helper.port_forward(node)
+    path = os.path.join(node.workspace.get_workspace(), ".monitor")
+
+
+def stop_node():
+    global conf
+    node = model.CNode(conf)
+    node.init()
+    node.stop()
+    node.terminate_workspace()
+    conf = {}
+
+
+class test_drive_capacity_consistency_1st(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        os.system("sudo rm /home/infrasim/.infrasim/test_drive_capacity.img")
+        start_node()
+
+    @classmethod
+    def tearDownClass(cls):
+        if conf:
+            stop_node()
+
+    def test_drive_capacity_consistency(self):
+        ssh = sshclient.SSH("127.0.0.1", "root", "root", port=2222)
+        assert ssh.wait_for_host_up() is True
+        s = UnixSocket(path)
+        s.connect()
+        s.recv()
+        # 1. expect: yml img zise == host disk size
+        r = os.popen("qemu-img info /home/infrasim/.infrasim/test_drive_capacity.img")
+        img_info = r.read()
+        r.close()
+        self.assertIn('virtual size: 1.0G', img_info, "Existing host drive image \
+                                                       size is different from the size defined in yaml")
+        # 2. expect: yml img size == guest disk size
+        status, stdout = ssh.exec_command('sg_readcap /dev/sdb')
+        self.assertIn('Device size: 1073741760 bytes', stdout, "Guest drive image \
+                                                                size is different from the size defined in yaml")
+        last_lba = re.search("Last logical block address=([0-9]{0,7})", stdout).group(1)
+        invalid_lba = int(last_lba) + 1
+        # 3. expect: report failed when access lba range out of 1G
+        cmd = "sg_dd if=/dev/sg1 bs=520 count=1 skip={0}".format(last_lba)
```

Comment:
shall we add some logic to make sure the /dev/sg1 is exactly the one you want access? from previous test, the generic number (sg1) might change in guest OS.